sdks/java/io/elasticsearch/src/main/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIO.java

```diff
@@ -2366,6 +2361,11 @@ protected BulkIOBaseFn(BulkIO bulkSpec) {
         this.spec = bulkSpec;
       }
 
+      @Override
+      public Duration getAllowedTimestampSkew() {
+        return Duration.millis(Long.MAX_VALUE);
```

Comment:
This won't affect the watermark, so it will just allow output to be dropped if it is assigned to an expired window.

Reply:
Elements should not be assigned to expired windows, because the elements are assigned to windows present in the active bundle. This is just a workaround the check for element output timestamp which is too restrictive. There is no need for elements not to be output with lower timestamp than timestamp of the current element, it is only needed not to output elements that change from on_time to late (or droppable). It would be best to relax the restriction, though it seems it would require a runner-specific code to be provided, because the check does not know about bundles.

Reply:
Yes the way in which this error arises is that multiple elements from the same bundle and window are buffered, and then later output.  But the timestamp of each element differs, even though they're in the same window and bundle.  The timestamp check only considers the output timestamp provided as arg against the timestamp of the _current_ element so in a case like this of buffering values which are all in the same window, the timestamp of the element at the point in time where the buffer has reached the desired size may result in checkTimestamp failures if any buffered elements from the same window had an earlier timestamp than the current element.

Agreed, it would be great to be able to relax the strictness of `checkTimestamp` if that can be done in a way that preserves the original intent of the method (like Jan said, ensuring "not to output elements that change from on_time to late (or droppable).")